### PR TITLE
[move-cli] showing coverage info for the whole test suite

### PR DESCRIPTION
**Pull request by mengxu-fb**
## Motivation

Currently, `move-cli test --track-cov` shows the line coverage information per every `args.txt` file. This might not be ideal when a Move developer have a test suite (i.e., many `args.txt` files) and we want to show the aggregated coverage information for the whole test suite instead of the information per each test. This PR tries to add the functionality of aggregating and showing coverage information for the whole test suite.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

`cargo xtest` for move-cli

## Related PRs

#6486

### CODE REVIEW

File: language/tools/move-cli/src/test.rs [hunk 1]
```diff
@@ -71,32 +73,71 @@ fn format_diff(expected: String, actual: String) -> String {
...
+            .insert(module_id.name().to_owned(), (entry, module));
     }
 
-    Ok(())
+    // collect filtered trace
+    let coverage_map = CoverageMap::from_trace_file(trace_file)
+        .to_unified_exec_map()
+        .into_coverage_map_with_modules(filter);
+
+    // done
```
Comment (sblackshear) [inline language/tools/move-cli/src/test.rs:131]: Nit: not very descriptive--kill?

File: language/tools/move-cli/tests/metatests/args.exp [hunk 2]
```diff
@@ -7,10 +7,10 @@ Command `test dummy`:
 Command `test cov`:
```
Comment (sblackshear) [inline language/tools/move-cli/tests/metatests/args.exp:7]: Should we expand this test to run multiple commands that hit multiple files to test the merging functionality?

**Review by sblackshear — COMMENTED** [review]
This LGTM but I defer to coverage master @tzakian on everything

File: language/tools/move-cli/src/test.rs [hunk 3]
```diff
@@ -111,6 +152,13 @@ pub fn run_one(args_path: &Path, cli_binary: &str, track_cov: bool) -> anyhow::R
             .output()?;
     }
     let mut output = "".to_string();
+
+    // for tracing file path: always use the absolute path so we do not need to worry about where
+    // the VM is executed.
+    let trace_file = env::current_dir()?
+        .join(&storage_dir)
```
[the remark this answers is not in the document]
Comment (mengxu-fb) [reply language/tools/move-cli/src/test.rs:159]: Yes, I'll switch to using `build_output`.

File: language/tools/move-cli/src/test.rs (hunk 1 above)
Comment (mengxu-fb) [reply language/tools/move-cli/src/test.rs:131]: Will do :)

File: language/tools/move-cli/tests/metatests/args.exp (hunk 2 above)
Comment (mengxu-fb) [reply language/tools/move-cli/tests/metatests/args.exp:7]: Yes, I am adding more test cases in an incoming commit.

Comment (mengxu-fb) [reply language/tools/move-cli/tests/metatests/args.exp]: Tests for the merging functionality added in commit #fab6008

**Review by sblackshear — APPROVED** [review]

**Pull request closed without merging**